src/lib/loader.js

```diff
@@ -130,6 +129,9 @@ export async function swapContent(url, isFirstRun) {
     });
   }
 
+  ga("set", "page", window.location.pathname);
```

Comment:
> Sets the new page as we change URL and explicitly call ga("set", "page", ...), which I think is needed (cc @philipwalton )

Not necessarily needed for the first pageview, but doing this helps get rid of unwanted query params in your analytics (which is generally what you want).

However, you're going to want to *also* do this when you call `pushState()` in your SPA router. This guide should have more details: https://developers.google.com/analytics/devguides/collection/analyticsjs/single-page-applications.

Reply:
This is where we load new content (i.e., related to calling `pushState`). `swapContent` is the callback into web.dev loading code from our router.

The question I had for you was more about calling `ga("set", "page", ...)`, which we _were not_ doing. I presume just triggering a pageview doesn't implicitly update the page, if GA sees a new location. Is that correct?

Reply:
Gah, sorry. For some reason I had it in my head that this was the template code (for the initial page load). Now I see that it's not.

> The question I had for you was more about calling ga("set", "page", ...), which we were not doing. I presume just triggering a pageview doesn't implicitly update the page, if GA sees a new location. Is that correct?

Yes, that's correct. You need to set the new `page` value, which you're doing. I took a second look at the staged PR and it looks correct to me.

You still might want to call `ga("set", "page", window.location.pathname);` in the template code before the first pageview to remove any query params from the URL that gets sent, but that's up to you.